client.go

```diff
@@ -359,11 +359,10 @@ func (c *Client) RPC(data []byte) ([]byte, error) {
 
 // Close closes Client connection and cleans up state.
 func (c *Client) Close() error {
-	err := c.Disconnect()
 	c.mutex.Lock()
 	c.status = CLOSED
 	c.mutex.Unlock()
-	return err
+	return c.Disconnect()
```

Comment:
looked more closely to this - we can't do in this way because as soon as client in CLOSED state some useful logic won't be executed inside `Disconnect()` - see `handleDisconnect` method - for example Disconnect and Unsubscribe events won't be fired. Though looks like you are right in a way that we should set CLOSED status immediately in `Close` method. Or maybe introduce new CLOSING status. So more internal refactoring actually required to fix this problem.